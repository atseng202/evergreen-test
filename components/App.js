// import BpkText from "bpk-component-text";

import Header from "./Header";

import styles from "../styles/App.module.scss";

import FlightCard from "./FlightCard";

const getClassName = (className) => STYLES[className] || "UNKNOWN";

/**
 * App Component
 * General container component for Header and FlightCards at "/" route
 *
 * Props:
 * itineraries, a list of itinerary objects like,
 *  [{ id, legs, price, agent, agent_rating}, ...]
 * legs, a list of leg objects like,
 *  [{ id, departure_aiport, arrival_airport, departure_time,
 *    arrival_time, stops, airline_name, airline_id, duration_mins}, ...]
 *
 * State: None
 *
 * Home -> App
 **/
function App({ itineraries, legs }) {
  function parseItineraries() {
    return itineraries.map((itinerary) => {
      // legs prop is a Set making it faster to find the
      // corresponding leg for each id
      const totalLegs = itinerary.legs.map((legId) => legs[legId]);

      const { id, price, agent } = itinerary;

      return (
        <FlightCard
          key={id}
          price={price}
          agent={agent}
          totalLegs={totalLegs}
        />
      );
    });
  }

  return (
    <div className={styles.App}>
      <Header />
      <main className={styles.App__main}>
        {/* <BpkText tagName="p">Over to you...</BpkText> */}
        {parseItineraries()}
      </main>
    </div>
  );
}

export default App;
