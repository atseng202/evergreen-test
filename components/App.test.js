import React from "react";
import { render } from "@testing-library/react";
import App from "./App";

describe("App smoke and snapshot tests", () => {
  let expectedProps;
  let legs = beforeEach(() => {
    expectedProps = {
      itineraries: [
        {
          id: "it_1",
          legs: ["leg_1", "leg_4"],
          price: "Â£35",
          agent: "Wizzair.com",
          agent_rating: 9.1,
        },
      ],
      legs: {
        leg_1: {
          id: "leg_1",
          departure_airport: "BUD",
          arrival_airport: "LTN",
          departure_time: "2020-10-31T15:35",
          arrival_time: "2020-10-31T17:00",
          stops: 0,
          airline_name: "Wizz Air",
          airline_id: "WZ",
          duration_mins: 145,
        },
        leg_4: {
          id: "leg_4",
          departure_airport: "LTN",
          arrival_airport: "BUD",
          departure_time: "2020-11-11T19:45",
          arrival_time: "2020-11-11T21:10",
          stops: 0,
          airline_name: "Wizz Air",
          airline_id: "WZ",
          duration_mins: 145,
        },
      },
    };
  });

  it("should render correctly", () => {
    render(<App {...expectedProps} />);
  });

  it("matches App snapshot", () => {
    const { asFragment } = render(<App {...expectedProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
