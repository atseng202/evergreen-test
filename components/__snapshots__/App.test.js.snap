// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App smoke and snapshot tests matches App snapshot 1`] = `
<DocumentFragment>
  <div
    class="App container"
  >
    <header
      class="Header"
    >
      <a
        href="/"
      >
        <span
          class="Header__hidden-text"
        >
          Skyscanner
        </span>
        <img
          alt="Skyscanner"
          class="Header__logo-image"
          src="/logo.svg"
        />
      </a>
    </header>
    <div
      class="App__main container"
    >
      <style
        type="text/css"
      >
        
        .btn-custom {
          background-color: #00a698;
          color: white;
        }
      
      </style>
      <div
        class="FlightCard card"
      >
        <div
          class="LegDetail container"
        >
          <div
            class="pt-2 pl-3 pr-3 my-auto row"
          >
            <div
              class="col-10"
            >
              <div
                class="LegDetail__dateInfo row"
              >
                <img
                  alt="WZ"
                  class="pr-2 my-auto LegDetail__airlineLogo"
                  src="https://logos.skyscnr.com/images/airlines/favicon/WZ.png"
                />
                <div
                  class="flex-grow-0 pr-4 col"
                >
                  <div
                    class="py-1 row"
                  >
                    15:35

                  </div>
                  <div
                    class="font-weight-light row"
                  >
                    BUD
                  </div>
                </div>
                <img
                  alt="arrow-right"
                  class="LegDetail__arrowRight"
                  src="/long-arrow-right.svg"
                />
                <div
                  class="flex-grow-0 pl-4 col"
                >
                  <div
                    class="py-1 row"
                  >
                    17:00

                  </div>
                  <div
                    class="font-weight-light row"
                  >
                    LTN
                  </div>
                </div>
              </div>
            </div>
            <div
              class="col"
            >
              <div
                class="justify-content-end py-2 font-weight-light row"
              >
                2h 25
              </div>
              <div
                class="justify-content-end LegDetail__noStops row"
              >
                Direct
              </div>
            </div>
          </div>
        </div>
        <div
          class="LegDetail container"
        >
          <div
            class="pt-2 pl-3 pr-3 my-auto row"
          >
            <div
              class="col-10"
            >
              <div
                class="LegDetail__dateInfo row"
              >
                <img
                  alt="WZ"
                  class="pr-2 my-auto LegDetail__airlineLogo"
                  src="https://logos.skyscnr.com/images/airlines/favicon/WZ.png"
                />
                <div
                  class="flex-grow-0 pr-4 col"
                >
                  <div
                    class="py-1 row"
                  >
                    19:45

                  </div>
                  <div
                    class="font-weight-light row"
                  >
                    LTN
                  </div>
                </div>
                <img
                  alt="arrow-right"
                  class="LegDetail__arrowRight"
                  src="/long-arrow-right.svg"
                />
                <div
                  class="flex-grow-0 pl-4 col"
                >
                  <div
                    class="py-1 row"
                  >
                    21:10

                  </div>
                  <div
                    class="font-weight-light row"
                  >
                    BUD
                  </div>
                </div>
              </div>
            </div>
            <div
              class="col"
            >
              <div
                class="justify-content-end py-2 font-weight-light row"
              >
                2h 25
              </div>
              <div
                class="justify-content-end LegDetail__noStops row"
              >
                Direct
              </div>
            </div>
          </div>
        </div>
        <div
          class="pt-3 container"
        >
          <div
            class="px-3 pb-3 justify-content-end row"
          >
            <div
              class="col"
            >
              <div
                class="font-weight-bold row"
              >
                <h2
                  class="mb-0"
                >
                  Â£35
                </h2>
              </div>
              <div
                class="FlightCard__agentText text-secondary font-weight-light row"
              >
                Wizzair.com
              </div>
            </div>
            <div
              class="col"
            >
              <div
                class="justify-content-end pr-2 pt-3 row"
              >
                <button
                  class="px-5 btn btn-custom btn-lg"
                  type="button"
                >
                  <h4
                    class="my-0"
                  >
                    Select
                  </h4>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
